### DISCUSSION TITLE
vi 54080000.vi: tegra_channel_error_status:error 4000

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#212] [Q&A]
Hi, 
the same camera driver code, which I tested in Nvidia Jetpack(4.6.1) and Yocto image(r32.6.1), in Nvidia Jetpack without any problem, but in the yocto image I get this error randomly. 
Does someone have idea? Many thanks!
[  +5.504766] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 323
[  +0.010939] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 326
[  +0.011669] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 327
[  +0.027073] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 328
[  +0.176216] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 336
[  +0.008547] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 339
[  +7.909677] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 811
[  +0.013990] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 814
[  +0.008594] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 815
[  +0.024170] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 816
[  +0.011398] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 817
[  +0.330701] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 834
[  +0.030137] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 837
[  +3.310245] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1033
[  +0.011153] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1036
[  +0.007940] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1037
[  +0.008616] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1038
[  +0.022319] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1039
[  +0.012201] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1040
[  +0.433352] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1064
[  +0.033104] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1067
[  +0.008397] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1069
[  +0.022633] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1070
[  +5.314889] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1386
[  +0.015588] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1389
[  +0.008200] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1390
[  +0.025992] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1391
[  +0.012990] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1392
[  +0.027335] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1393
[  +0.134684] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1399
[  +0.008994] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 1402
[ +18.240584] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2494
[  +0.012088] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2497
[  +0.010495] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2498
[  +0.023996] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2499
[  +0.014485] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2500
[  +0.254508] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2512
[  +0.008330] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2515
[  +4.276389] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2767
[  +0.007530] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2770
[  +0.024126] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2771
[  +0.008575] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2772
[Sep20 10:48] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2783
[  +0.018917] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 2786
[  +4.339508] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3043
[  +0.008507] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3046
[  +0.026125] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3047
[  +0.009472] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3048
[  +0.029929] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3049
[  +0.107592] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3053
[  +0.009297] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3056
[  +2.061126] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3178
[  +0.010532] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3181
[  +0.008783] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3182
[  +0.034177] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3183
[  +0.010365] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3184
[  +0.029124] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3185
[  +0.222668] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3195
[  +0.007349] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3198
[  +0.009580] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 3199
[ +14.440909] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4063
[  +0.062984] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4067
[  +0.170522] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4076
[  +2.293817] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4212
[  +0.008771] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4215
[  +0.022856] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4216
[  +0.009250] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4217
[  +0.053486] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4218
[  +0.254865] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4229
[  +0.007745] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4232
[  +0.008885] vi 54080000.vi: tegra_channel_error_status:error 4000 frame 4233

Best regards
Danny

**Comment by madisongh** [comment]
Have you searched around for that error?  I see several mentions in the NVIDIA developer forums, which you might find helpful.  That 0x4000 code indicates an error from the CSI pixel parser.... not being a camera expert, I'm not sure exactly what that means, but perhaps some kind of timing issue with the delivery of the frame over the CSI interface?

There are several environmental differences between the default JetPack Ubuntu setup (which is essentially a kitchen-sink environment that caters specifically to developers) and the setup here. For example, I believe the JetPack setup runs the `jetson_clocks` script to boost the clocks on the chip  at boot time, which is not done here.  If you include the `tegra-tools-jetson-clocks` package in your build (some of the demo images do that, but not all), you could try running that before testing your camera and see if that makes a difference.

If it does, then you still might want to review your hardware and driver and device tree configuration for the sensor.  If it only works properly when the clocks are ramped up to maximum, that's probably an indication that something's not right there.  I have no problem using Raspberry Pi V2 (IMX219) cameras on Nano devkits with the default clock settings.

**Reply by danny1209** [reply]
Hi @madisongh 
I tried jetson_clocks, the output is:  nvgpu: 57000000.gpu             railgate_enable_store:297  [INFO]  railgate is disabled.
but it has no impact on "tegra_channel_error_status:error 4000 frame xxx". it´s same like before. 

In the meanwhile I also tried two kinds of fakra cables(FPD LINK 3), with one of them I see obviously less error outputs, but still some. 
Why does the cable make the difference?

So the same hardware and driver, which work on Jetpack, but not on yocto, there should are some diffferences between Jetpack and Yocto.

**Reply by danny1209** [reply]
Hi @madisongh 

when freqeuncy range of the FPD LINK III cable: 0-3G, I got this kind of error.
When 0-6G, no problem.
![image](https://user-images.githubusercontent.com/46848895/184882010-3144805e-d2a0-4b67-825a-cb227cfe8d04.png)

could you image how it makes difference?
